tests/test_worker.py

```diff
@@ -247,6 +248,23 @@ def test_heartbeat(self):
         self.testconn.hdel(w.key, 'birth')
         w.refresh()
 
+    @slow
+    def test_heartbeat_busy(self):
+        """Periodic heartbeats while horse is busy with long jobs"""
+        q = Queue()
+        w = Worker([q])
+
+        for timeout, expected_heartbeats in [(3, 0), (35, 1), (65, 2)]:
```

Comment:
This test would take about 100 seconds to run, which is a long time (the CI builds will take a lot of resources). We should expose `Worker.job_monitoring_interval` that allows us to customize this monitoring interval and change it to (for example 2 seconds) during tests.